MQTT define bugfix

brunnels opened the pull request:
Possilbe mDNS bugfix
Switched to DBG method instead of multiple ifdef in httpd
Added windows.md

brunnels committed:
MQTT define bugfix
Possilbe mDNS bugfix
Switched to DBG method instead of multiple ifdef in httpd

tve commented on WINDOWS.md:
```diff
@@ -0,0 +1,13 @@
+* Install [SourceTree](https://www.sourcetreeapp.com) and check CLI git or other git distribution to do git from CLI
+* Install  and latest Java
```
Is this missing a word?

brunnels committed:
Fixed WINDOWS.md

tve commented on esp-link/cgiwifi.c:
```diff
@@ -121,10 +121,11 @@ static ICACHE_FLASH_ATTR
   if (!mdns_started) {
     os_memset(&mdns_info, 0, sizeof(struct mdns_info));
-    mdns_info.host_name = flashConfig.hostname;
+    mdns_info.host_name = flashConfig.hostname,
```
why a comma?

brunnels replied:
That's the possible bug fix.  All the other lines after have a comma as well so I assumed it should too.  I don't know if they were wrong or it is wrong.  The other implementations and examples I've found allocate the memory differently too so that might be the stability issue.  Here's it is mimicking other implementations:

``` c
struct mdns_info *info = (struct mdns_info *)os_zalloc(sizeof(struct mdns_info));
info->host_name = flashConfig.hostname;
info->ipAddr = ip.addr;
info->server_name = "http";
info->server_port = 80;
info->txt_data[0] = (char *) "version = now";
espconn_mdns_init(info);
```

tve replied:
Let's try the zalloc version to see whether that fixes the issues I'm seeing with mdns..

tve commented on esp-link/cgiwifi.c:
```diff
@@ -121,10 +121,11 @@ static ICACHE_FLASH_ATTR
     mdns_info.server_port = 80,     // service port
-    mdns_info.ipAddr = ip.addr,
+    mdns_info.ipAddr = ip.addr,  
+    mdns_info.txt_data[0] = (char *) "version = now",
```
what does that do?

brunnels replied:
all the other implementations I've seen define something for the txt_data

brunnels committed:
Moved to debug macro for SLIP_DBG and fixed an error when SLIP_DBG was defined
Removed unnecessary stuff
Switched VS project to use sdk 1.4.0 sources for intellisense

brunnels committed:
Started adding in standalone mdns implementation

brunnels committed:
Updates for SDK 1.5

brunnels committed:
renamed debug command for consistancy

brunnels committed:
Fixed bug where callbacks were not getting called if the CMD MQTT client called connect while the ESP MQTT client was connected or disconnected.

brunnels committed:
Updated DBG macros as suggested by susisstrolch

brunnels committed:
DOH! missing semicolon

brunnels committed:
Added in sntp update

brunnels committed:
Merge branch 'master' of https://github.com/jeelabs/esp-link
Added services page and moved some things around

brunnels committed:
fixes for services

brunnels committed:
Removed unused mdns files

tve commented on esp-link/cgiflash.c:
```diff
@@ -150,10 +148,10 @@ static ETSTimer flash_reboot_timer;
+  if (!canOTA()) {
+    errorResponse(connData, 400, flash_too_small);
+    return HTTPD_CGI_DONE;
+  }
```
Looks like the indentation is off with your changes in this file?

brunnels replied:
Strange.  I'll fix that.

brunnels replied:
Actually that whole file is 4 space indented rather then 2.  You want it 4 space indented or normalized to 2 space indention?

tve replied:
I don't care much as long as it's consistent..

tve commented on esp-link/cgiflash.c:
```diff
@@ -190,3 +184,18 @@ int ICACHE_FLASH_ATTR cgiRebootFirmware(HttpdConnData *connData) {
+  httpdEndHeaders(connData);
+
+  // Schedule a reboot
+  system_upgrade_flag_set(UPGRADE_FLAG_FINISH);
```
Why are you setting this?

brunnels replied:
No reason.  Inadvertently copied from the flash reboot method.  I'll remove it.

tve replied:
Ooops, never mind, github's diff made it look wrong

susisstrolch commented on esp-link/config.c:
```diff
@@ -20,20 +20,17 @@ FlashConfig flashDefault = {
+  .rx_pullup	  = 1,
+  .mdns_enable  = 1, .mdns_servername = "http\0",
+  .sntp_server  = "us.pool.ntp.org\0", .timezone_offset = 0,
+  .syslog_enable= 0, .syslog_host = "\0", .syslog_minheap = 8192, .syslog_filter = 7, .syslog_showtick = 1, .syslog_showdate = 0,
```
IMHO, syslog_enable (and it's handling) isn't necessary. By setting syslog_host to "", syslog gets disabled by syslog_init(). You can also (re)use the syslog_filter to avoid sending syslog messages.

susisstrolch commented on html/wifi/wifi.html:
```diff
@@ -30,8 +30,7 @@
-                style="width:auto; display:inline-block; margin-left: 0.7em">
+              <input type="text" id="hidden-ssid" value="" style="width:auto; display:inline-block; margin-left: 0.7em">
             </label>
             <label>WiFi password, if applicable:</label>
```
change to Wifi to stay consistent with your mod. from line 13

brunnels replied:
I just fixed the source indentation but I see the "WiFI" on line 12 and "Wifi" on line 13 now that you pointed it out.  Looks like a mixed bag of "WiFI" and "Wifi" throughout so I'l normalize everything to WiFI.

brunnels committed:
Normalized Wifi to WiFI.
Fixed button spacing on console clear and log reset buttons
Renamed cgiReboot method to cgiReset and called the proper reset method
Normalized whitespace in cgiflash.c

susisstrolch commented on esp-link/config.c:
```diff
@@ -20,20 +20,17 @@ FlashConfig flashDefault = {
-  .syslog_showdate= 0,
+  .sys_descr 	  = "\0",
+  .rx_pullup	  = 1,
+  .mdns_enable  = 1, .mdns_servername = "http\0",
```
please, move your config values to the end, otherwise you will break each ones config who already build from master

brunnels committed:
Fixed bug with services page submitting on field click
Fixed bug with Makefile when COMPRESS_W_HTMLCOMPRESSOR = no

brunnels committed:
removed syslog_enable functionality and config
reordered config settings

brunnels committed:
Fixed mdns settings bug

brunnels committed:
Fix for services page syslog filter select list
Fix for a couple compiler errors when cycling the different undef and define options in user_config.h

brunnels committed:
Switched from event.srcElement to event.target for better browser compatibility

brunnels committed:
replaced the other srcElement too
